tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -261,6 +279,23 @@ def test_download_and_prepare_as_dataset(self):
     else:
       self._download_and_prepare_as_dataset(self.builder)
 
+    if not self.SKIP_CHECKSUMS:
+      with self._subTest("url_checksums"):
+        self._test_checksums()
+
+  def _test_checksums(self):
+    urls = []
+    # how to get the path to the checksum file corresponding to the testcase.
+    filepath = os.path.join(download.checksums._get_path(os.path.basename(self.example_dir)))
```

Comment:
@Conchylicultor I am unable to get the file path to the correct checksum file.
The above code does not work for testcases like `translate/wmt_test.py`
and
`self.builder.name`, `self.info.name` does not work in for `sun_test.py`.

Reply:
Why don't you use `checksums._get_path(dataset_name)` ? It will already raise the error if it don't find the file

Reply:
I agree we could this but for test files like sun_test.py, I get the following error.
```
File "/home/vijay/GitHub-Repos/Tensorflow/datasets/tensorflow_datasets/testing/dataset_builder_testing.py", line 289, in _test_checksums
filepath = os.path.join(download.checksums._get_path(self.dataset_name))
AttributeError: 'Sun397TfdsTest' object has no attribute 'dataset_name'

```

Reply:
or if we use `checksums._get_path(self.builder.name)`. I got the following error
```
AssertionError: No checksums file could be found for dataset testable_sun397. Please create one in one of:
```
So, is there any proper way to handle all these errors and get the path to proper checksum file.

Reply:
What is the content of `checksums._CHECKSUM_DIRS` ?
Note that when encountering a bug like this, you should try to look at the code which raise the error and try to debug from there.

Reply:
Also it is expected that this fail if you haven't created a `testable_sun397.txt` file first. Why don't you try to test on an existing dataset test ?